crates/router/src/connector/klarna/transformers.rs

```diff
@@ -73,13 +83,40 @@ pub struct KlarnaPaymentsRequest {
     shipping_address: Option<KlarnaShippingAddress>,
 }
 
+#[derive(Default, Debug, Serialize, Deserialize)]
+pub struct CheckoutRequest {
```

Comment:
the fields in this and PaymentsRequest seem to be similar. Create an enum for the fields that differ and you can flatten them. You can take reference from how it is done for other connectors.

Reply:
Can you provide me example of connectors that is using the same approach you suggested, I couldn't find it.

Reply:
https://github.com/juspay/hyperswitch/blob/8874a1d450555a04119f4b975132c01956ed3fd7/crates/router/src/connector/stripe/transformers.rs#L137